2-mandatory/2-water-bottle.js

```diff
@@ -18,45 +19,56 @@ You have to implement the missing features according to the specification.
 let bottle = {
   volume: 0,
   fillUp: function () {
+    this.volume = 100;
     // calling this function should completely fill your bottle (volume = 100);
   },
   pour: function () {
+    if(this.volume <= 90){
+       this.volume += 10;
+    }
     // calling this function should increase your bottle volume by 10 units;
   },
   drink: function () {
+    if(this.volume >= 10){
+     this.volume -= 10;
+    }
     // calling this function should decrease your bottle volume by 10 units;
   },
   isFull: function () {
+    return this.volume === 100 ? true : false;
+
     // this function should return true if your bottle is full;
   },
   isEmpty: function () {
+    return this.volume === 0 ? true : false;
     // this function should return true if your bottle is empty;
   },
 };
 
 /*
 TIP:
   Remember that for changing properties on the current object inside one of its
-  methods you can refer to it by its variable name: `bottle` or by using the keyword `this`.
+  methods you can refer to it by its variable name: bottle or by using the keyword this.
 */
 
 /*
 Extra question:
-  Why do you think it is preferred to use `this` inside the object rather than its variable name, in our case `bottle`?
+  Why do you think it is preferred to use this inside the object rather than its variable name, in our case bottle?
   Leave your answer below:
 */
 
 // Write you answer to the question here
+// Keyword 'this' makes functions reusable, function are independent from variable's name.
```

Comment:
The main difference why we use `this` instead of variable name is that they will give a different result. 
using this inside the object/class will give you access to the current instance of that class
using the variable name will give you _a reference_ to the class

Here's a bit of discussion and examples https://stackoverflow.com/questions/50558575/why-do-you-have-to-use-keyword-this-instead-of-class-name